TodoListSPA/app.js

```diff
@@ -0,0 +1,238 @@
+'use strict';
+
+config.callback = loggedin;
+var authenticationContext = new AuthenticationContext(config);
+
+if (!config.popUp) {
+    if (authenticationContext.isCallback(window.location.hash)) {
+        authenticationContext.handleWindowCallback();
+    }
+    else {
+        var user = authenticationContext.getCachedUser();
+        if (user && window.parent === window && !window.opener) {
+            // Display the user
+            displayUserAndShowSignOutButton(user);
+
+            // Call the protected API to show the content of the todo list
+            acquireAnAccessTokenAndCallTheProtectedService();
```

Comment:
This function name is a bit verbose and wraps two concepts.  I think in the top level function it makes sense to break apart the concepts of getting a token from using it for teaching purposes. In general, we don't encourage developers to wrap AcquireToken functions as they're already abstracted to a top-level function.